### PR TITLE
Disable example generation and modify code generation to be decoupled from docs

### PR DESCRIPTION
**Pull request by karencfv**
~_Update: DO NOT MERGE!!!! Wait for other clients to be updated as well. See below comments._~

Heya @jessfraz! Could you take a look at this one? Just want to make sure I didn't leave any loose ends here :)

Related: https://github.com/oxidecomputer/docs/issues/217

### CODE REVIEW

**Comment by jessfraz** [comment]
Wait so I wouldn’t break our existing system til we have another that works. Does that make sense? Otherwise if the path we choose doesn’t work out then we can’t go back

**Comment by jessfraz** [comment]
Can we prove out another system in branches first?

**Comment by jessfraz** [comment]
And to be clear it’s not that I don’t think it will work it’s just I would like to treat the docs as “working” right and if we prove this all out in branches there will be zero times of the docs “not working” as in not having up to date code samples that actually reflect reality

**Comment by karencfv** [comment]
Sounds good @jessfraz! I won't merge this one until I have the other clients modified as well. 

Just worth mentioning that this won't affect the docs site anymore, given that David and I worked on this yesterday  https://github.com/oxidecomputer/docs/pull/227 as the first part of https://github.com/oxidecomputer/docs/issues/217

File: rust-toolchain [hunk 1]
```diff
@@ -1 +1 @@
-stable
+nightly
```
Comment (david-crespo) [inline rust-toolchain:1]: what's this about?

**Comment by jessfraz** [comment]
That sounds fine

File: rust-toolchain (hunk 1 above)
Comment (karencfv) [reply rust-toolchain:1]: Some of the features that the generator requires only work with the nightly toolchain. I was getting these warnings when generating:
```
Warning: can't set `format_code_in_doc_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `format_strings = true`, unstable features are only available in nightly channel.
Warning: can't set `imports_granularity = Crate`, unstable features are only available in nightly channel.
Warning: can't set `group_imports = StdExternalCrate`, unstable features are only available in nightly channel.
```

So this is just a little bug fix :)

**Review by david-crespo — APPROVED** [review]

**Pull request merged**